raster/r.what/main.c

```diff
@@ -278,6 +304,17 @@ int main(int argc, char *argv[])
 
         fprintf(stdout, "\n");
     }
+    else if (strcmp(opt.format->answer, "json") == 0) {
+        i = 0;
+        names = G_malloc(nfiles * sizeof(char *));
+
+        ptr = opt.input->answers;
+        for (; *ptr != NULL; ptr++) {
+            names[i] = G_malloc(GNAME_MAX);
+            strcpy(names[i], *ptr);
+            i++;
+        }
```

Comment:
Is this whole thing needed?

Reply:
I saw in the plain case to write the header, the ptr is being copied first to `char name[GNAME_MAX]`, so I similarly created a `names` array. Should I directly access the `opt.input->answers` pointer instead?

Reply:
Hmm, I am not sure why it is used there, seems using `answers` directly should work.

Reply:
Done.